### PR TITLE
[ENG-1899] Added SignString Class

### PR DESCRIPTION
**Pull request by WarrickFitz**

### CODE REVIEW

**Review by dhruv-kumar-jha — APPROVED** [review]

File: src/interfaces/signString.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { PrivateKey, SignStringOptions, SignStringSignResult, SignStringValidateResult } from '../models'
+/**
+ * The Transaction interface declares the operations that all concrete chain (chain)transaction classes must implement
+ */
+export interface SignString {
```
Comment (traylewin) [inline src/interfaces/signString.ts:5]: Let's call this SignMessge instead

File: src/models/signStringModels.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/** SignString options used when contructing a signString object */
+export type SignStringOptions = {
```
Comment (traylewin) [inline src/models/signStringModels.ts:2]: Rename all to SignMessage...

**Review by traylewin — COMMENTED** [review]
We need to add this SignMessage functionality to EOS and Algorand plugins too

**Review by traylewin — COMMENTED** [review]
We also need to update chain-js-playground to show example signMessage

File: src/interfaces/signString.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { PrivateKey, SignStringOptions, SignStringSignResult, SignStringValidateResult } from '../models'
+/**
+ * The Transaction interface declares the operations that all concrete chain (chain)transaction classes must implement
```
[the remark this answers is not in the document]
Comment (artmanque) [reply src/interfaces/signString.ts:3]: *accurate

File: src/models/signStringModels.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  signMethod: string
```
[the remark this answers is not in the document]
Comment (GaetanoMondelli) [reply src/models/signStringModels.ts]: Discussed during call

File: src/models/signStringModels.ts (hunk 2 above)
Comment (GaetanoMondelli) [reply src/models/signStringModels.ts]: At the moment, I have renamed files, classes, and fields. If we determine that the term "string" is more abstract and suitable for this context, I am open to reverting the changes back.

File: src/interfaces/chain.ts [hunk 5]
```diff
@@ -147,6 +147,8 @@ export interface Chain {
   supportsGetPublicKeyFromSignature: boolean
   /** Whether the chain supports resources */
   supportsResources: boolean
+  /** Whether the chain suppports ERC712 data type signature*/
+  supportsTypedDataSignature: boolean
```
Comment (traylewin) [inline src/interfaces/chain.ts:151]: Please rename this member to supportsSignMessageTypedData

File: src/interfaces/chain.ts [hunk 6]
```diff
@@ -147,6 +147,8 @@ export interface Chain {
   supportsGetPublicKeyFromSignature: boolean
   /** Whether the chain supports resources */
   supportsResources: boolean
+  /** Whether the chain suppports ERC712 data type signature*/
```
Comment (traylewin) [inline src/interfaces/chain.ts:150]: Please tweak the comment - this is generic for all chains - so change to something like:
/** Whether the chain's signMessage feature supports signing a typed data object (ex: ERC712 data type for Ethereum)*/

**Review by traylewin — COMMENTED** [review]
Please remove the signMessage member from the chain interface - and for all plugins as well

**Review by traylewin — APPROVED** [review]
Approved with changes requested. Please note that there are a few minor changes that affect all three plugins

**Review by traylewin — COMMENTED** [review]
tweak comment

**Review by traylewin — APPROVED** [review]

**Pull request merged**